Automattic/wp-calypso
# Hosting Command Palette: Add tests for rendering command palette

Related to https://github.com/Automattic/dotcom-forge/issues/4952

## Proposed Changes

This PR adds unit tests to check the rendering of the command palette. Here are some of the test cases that it covers:

* it confirms that the command palette opens with the commands added to the commands array;
* it confirms that the command palette closes when Escape key is pressed;
* it confirms that the command palette closes when clicking outside the modal overlay;
* it confirms that the command palette closes when you chose a command that does not have a callback set;
* it confirms that the command palette displays only the commands that match label and searchLabel when the search is set;
* it confirms that the command palette displays `Not results found` message when there is no match in search;
* it confirms that command palette can navigates to a nested command and back to the root commands if you select a command with a callback

## Testing Instructions

* Pull the changes from this PR locally
* Run `yarn test-client -- client/components/command-palette/test/index.tsx` in the terminal
* Confirm that the tests are passing

## Pre-merge Checklist
